HEN_HOUSE/egs++/geometry/egs_autoenvelope/egs_autoenvelope.cpp

```diff
@@ -0,0 +1,1018 @@
+        return tmin;
+    }
+    vector<EGS_BaseGeometry *> geoms_in_reg = getGeomsInRegion(ireg);
+    for (vector<EGS_BaseGeometry *>::iterator geom=geoms_in_reg.begin(); geom!= geoms_in_reg.end(); ++geom) {
+        EGS_Float local_t = (*geom)->hownear(-1, x);
+        if (local_t < tmin) {
+            tmin = local_t;
+            if (tmin < 0) {
+                return tmin;
+            }
+        }
+    }
+    return tmin;
+};
+
+bool EGS_AEnvelope::hasBooleanProperty(int ireg, EGS_BPType prop)  {
+    if (ireg >= 0 && ireg < nreg) {
+        if (ireg < nregbase) {
+            return base_geom->hasBooleanProperty(ireg,prop);
+        }
+        volcor::GeomRegPairT local = getLocalFromGlobalReg(ireg);
+        return local.first->hasBooleanProperty(local.second, prop);
+    }
+    return false;
+};
+
+void EGS_AEnvelope::setBooleanProperty(EGS_BPType prop) {
+    setPropertyError("setBooleanProperty()");
+};
+
+void EGS_AEnvelope::addBooleanProperty(int bit) {
+    setPropertyError("addBooleanProperty()");
+};
+
+void EGS_AEnvelope::setBooleanProperty(EGS_BPType prop, int start, int end,int step) {
+    setPropertyError("setBooleanProperty()");
+};
+
+void EGS_AEnvelope::addBooleanProperty(int bit,int start,int end,int step) {
+    setPropertyError("addBooleanProperty()");
+};
+
+int EGS_AEnvelope::getMaxStep() const {
+    int nstep = base_geom->getMaxStep();
+    for (size_t j=0; j< inscribed_geoms.size(); ++j) {
+        nstep += inscribed_geoms[j]->getMaxStep();
+    }
+    return nstep + inscribed_geoms.size();
+};
+
+EGS_Float EGS_AEnvelope::getMass(int ireg) {
+
+    if (ireg < 0) {
+        return -1;
+    }
+    else if (ireg < nregbase) {
+        return corrected_mass[ireg];
+    }
+
+    volcor::GeomRegPairT local = global_reg_to_local[ireg];
+
+    return local.first->getMass(local.second);
+};
+
+EGS_Float EGS_AEnvelope::getMassCorrectionRatio(int ireg) {
+
+    if (0 <= ireg && ireg < nregbase) {
+        return corrected_mass[ireg]/uncorrected_mass[ireg];
+    }
+
+    return 1;
+};
+
+
+/* Print information about the geometry. If `print debug info = yes` is present
+ * in the geometry, extra information about which regions of the base geometry
+ * contain inscribed geometries and how their volumes were corrected.
+*/
+void EGS_AEnvelope::printInfo() const {
+
+    EGS_BaseGeometry::printInfo();
+
+    if (debug_info) {
+
+        for (int ir=0; ir < nregbase; ir++) {
+            if (fabs(uncorrected_mass[ir] - corrected_mass[ir]) < 1E-8) {
+                egsInformation("    mass of region %d was corrected from %.5E g to %.5E g\n",
+                               ir, uncorrected_mass[ir], corrected_mass[ir]);
+            }
+        }
+
+        for (int ir=0; ir < nregbase; ir++) {
+            if (geoms_in_region[ir].size() > 0) {
+                egsInformation("    region %d has %d insribed geometries\n", ir, geoms_in_region[ir].size());
+            }
+        }
+    }
+
+    vc_results.outputResults();
```

Comment:
I get a segmentation fault around here if you load volumes from a volume correction file (used "text" mode). I think we'll need to do some robustness testing around the `volume correction file` option. If you comment this line out, there is no segmentation fault.